vllm/v1/engine/core.py

```diff
@@ -162,6 +162,11 @@ def _initialize_kv_caches(
         elapsed = time.time() - start
         logger.info(("init engine (profile, create kv cache, "
                      "warmup model) took %.2f seconds"), elapsed)
+
+        # Log GPU load time if using CUDA
+        if vllm_config.device_config.device_type == "cuda":
+            self.log_gpu_load_time(vllm_config, elapsed)
```

Comment:
Let's make this something that other devices can implement without any code changes here - e.g.

```
def log_startup_stats(self, engine_init_time):
    stats = self.model_executor.get_startup_stats()
    if stats is None:
        return
    ....
```